http_2.rflx

```diff
@@ -0,0 +1,230 @@
+			Settings_Value_Enable_Push : Settings_Value_Enable_Push
+				then null;
+			Settings_Value_Initial_Window_Size : Settings_Value_Initial_Window_Size
+				then null;
+			Settings_Value_Max_Frame_Size : Settings_Value_Max_Frame_Size
+				then null;
+			Settings_Value_Generic : Settings_Value_Generic
+				then null;
+			Pad_Length : Pad_Length
+				then Data
+					with Size => (Payload_Length - (Pad_Length + 1)) * 8 	-- 1 -> Pad_Length'Length
+					if Frame_Type = DATA and Payload_Length >= Pad_Length + 1
+				then Exclusive_Flag
+					if Frame_Type = HEADERS and Flag_Priority = True
+				then Header_Block_Fragment
+					with Size => Payload_Length * 8
+					if Frame_Type = HEADERS and Flag_Priority = False
+				then Reserved_Bit_2
+					if Frame_Type = PUSH_PROMISE;
+			Reserved_Bit_2 : Zero_Bit
+				then Promised_Stream_ID
+					if Frame_Type = PUSH_PROMISE
+				then Last_Stream_ID 
+					if Frame_Type = GOAWAY
+				then Window_Size_Increment
+					if Frame_Type = WINDOW_UPDATE;
+			Promised_Stream_ID : Stream_ID
+				then Header_Block_Fragment
+					with Size => (Payload_Length - (Pad_Length + 5)) * 8
+					if Flag_Padded = True and Payload_Length >= Pad_Length + 5
+				then Header_Block_Fragment
+					with Size => (Payload_Length - 4) * 8
+					if Flag_Padded = False and Payload_Length >= 4;
+			Last_Stream_ID : Stream_ID
+				then Error_Code;
+			Exclusive_Flag : Boolean
+				then Stream_Dependency;
+			Stream_Dependency : Stream_ID
+				then Weight;
+			Weight : Weight
+				then Header_Block_Fragment
+					with Size => (Payload_Length - (Pad_Length + 6)) * 8
+					if Frame_Type = HEADERS and Flag_Padded = True and Payload_Length >= Pad_Length + 6
+				then Header_Block_Fragment
+					with Size => (Payload_Length - 5) * 8				 -- Sub prev field lengths of frame type
+					if Frame_Type = HEADERS and Flag_Padded = False and Payload_Length >= 5
+				then null
+					if Frame_Type = PRIORITY;
+			Header_Block_Fragment : Opaque
+				then Padding
+					with Size => Pad_Length * 8
+					if (Frame_Type = HEADERS or Frame_Type = PUSH_PROMISE) and Flag_Padded = True
+				then null
+					if Frame_Type = CONTINUATION or
+					 	((Frame_Type = PUSH_PROMISE or
+					 	Frame_Type = HEADERS) and Flag_Padded = False);
+			Data : Opaque
+				then Padding
+					with Size => Pad_Length * 8
+					if Frame_Type = DATA and Flag_Padded = True
+				then null
+					if (Frame_Type = DATA and Flag_Padded = False or Frame_Type = PING);
+			Padding : Opaque
+				then null;
+			Error_Code : Error_Code
+				then null
+					if Frame_Type = RST_STREAM
+				then null
+					if Frame_Type = GOAWAY and Payload_Length <= (Error_Code'Size + Last_Stream_ID'Size + Reserved_Bit_2'Size)
+				then Additional_Debug_Data
+					with Size => (Payload_Length - (Error_Code'Size + Last_Stream_ID'Size + Reserved_Bit_2'Size)) * 8
+					if Frame_Type = GOAWAY and Payload_Length > (Error_Code'Size + Last_Stream_ID'Size + Reserved_Bit_2'Size);
```

Comment:
I suppose `Payload_Length` defines the length in bytes, not in bits.

Reply:
Fixed in e3a85bb